### PR TITLE
After deployment fixes August 4, 2022

### PR DESCRIPTION
**Pull request by saas786**
* https://app.clickup.com/t/2rh69b1
* https://sentry.io/organizations/cxl/issues/3477686498/events/?project=1300539&referrer=slack

### CODE REVIEW

File: includes/class-sensei-lesson.php [hunk 1]
```diff
@@ -3174,12 +3174,12 @@ public function lesson_quiz_questions( $quiz_id = 0, $post_status = 'any', $orde
 						// If this is a multiple question then get the specified amount of questions from the specified category.
 						$question_cat    = (int) get_post_meta( $question->ID, 'category', true );
 						$question_number = (int) get_post_meta( $question->ID, 'number', true );
+						$quiz_author     = get_post( $quiz_id )->post_author;
```
Comment (pawelkmpt) [inline includes/class-sensei-lesson.php:3177]: Is this taken form upstream? I see the risk of post not being present and code crashing.

**Review by pawelkmpt — COMMENTED** [review]
First commit fixes reported issue. What about other commits? How do they correspond with the first one?

Also links to upstream solutions are missing.

File: includes/class-sensei-lesson.php (hunk 1 above)
Comment (saas786) [reply includes/class-sensei-lesson.php:3177]: > Is this taken form upstream? I see the risk of post not being present and code crashing.

Yes.

Ref: https://github.com/Automattic/sensei/blob/3bd0d7823742e787e914e123799860ae01f44cf6/includes/class-sensei-lesson.php#L3528

**Comment by saas786** [comment]
> First commit fixes reported issue. What about other commits? How do they correspond with the first one?
> 
> Also links to upstream solutions are missing.

@pawelkmpt 
Issue was from `/includes/class-sensei-lesson.php`, so fixed it in one commit, but after comparing with upstream version, I noticed there were some changes we can bring in, which can avoid further issue from same file, so I added additional commits.

Tracking down each commit for change(s), its gonna take some time, as when I filled last round of `sync with upstream`, I had to spend couple of hours just to find a commit which we can use, as there are like 8k commits between our fork and upstream repo.

If you want, I can do that, and do `cherry pick` and implement changes according?

**Comment by pawelkmpt** [comment]
> @pawelkmpt Issue was from `/includes/class-sensei-lesson.php`, so fixed it in one commit, but after comparing with upstream version, I noticed there were some changes we can bring in, which can avoid further issue from same file, so I added additional commits.

My point is: hotfix branch is supposed to fix some urgent problem. Do other commits fix this problem as well? If not, they should be in separate branch or at least properly described, so I don't have to wonder why suddenly phpdoc changes are in hotfix branch.

> Tracking down each commit for change(s), its gonna take some time, as when I filled last round of `sync with upstream`, I had to spend couple of hours just to find a commit which we can use, as there are like 8k commits between our fork and upstream repo.
> 
> If you want, I can do that, and do `cherry pick` and implement changes according?

I think we discussed that in the past and I noted that we need reference to original commit and you've beautifully done in recently.

`git blame` doesn't help with the process? What is your process for choosing what to bring into our repo? Maybe it needs adjustment? Or maybe it's so time consuming it's just not worth it?

cc @lkraav

**Comment by saas786** [comment]
> `git blame` doesn't help with the process? What is your process for choosing what to bring into our repo? Maybe it needs adjustment? Or maybe it's so time consuming it's just not worth it?

I review relevant file `history` and see look for relevant commits.
No didn't use `git blame`.

Found one commit relevant to changes, going to `cherry pick` it.

- https://github.com/Automattic/sensei/commit/4b76db38dca2ce7cfcfa0f12c00b57d2904e9ace

**Comment by pawelkmpt** [comment]
@saas786 I see you keep working in this PR, over 130 updates but...
* what happened to hotfix?
* why Sensei sync happens in hotfix branch?

**Comment by saas786** [comment]
> @saas786 I see you keep working in this PR, over 130 updates but...
> 
> * what happened to hotfix?

:) Initially I had a plan to just fix `undefined $quiz_author` issue, but when you asked for relevant commits where this change happened, I noticed there were several issues which can / might have been happening, so started bring those fixes. But than finally I saw I can at least bring `includes/class-sensei-lesson.php` upto date with `upstream v4.6.1`, so we have less friction moving forward.

Note: You can just deploy these commits for hot fix issue.

- [conversionxl/sensei@`5bd9528` (#9)](https://github.com/conversionxl/sensei/pull/9/commits/5bd952834dcc4d0b91e48b2655fa146ebc408a52)
- [conversionxl/sensei@`ac74ac3` (#9)](https://github.com/conversionxl/sensei/pull/9/commits/ac74ac3bfef811712bd1d35c4b70c2f7c7b0ad95)
- [conversionxl/sensei@`2c03c3f` (#9)](https://github.com/conversionxl/sensei/pull/9/commits/2c03c3f7dd0d1b983485a5fe0e81f693dbc0966b)

> * why Sensei sync happens in hotfix branch?

:) Was splitting it into two separate branches, but would have added some review and deployment friction, so kept it as is.

---

@lkraav @pawelkmpt 

While we are on the subject, do you think I should merge irrelevant files as is (upto 4.6.1), in single commit, as bring their commit history would be hard to review, to keep commits minimal moving forward?
As it would make it easier to compare & bring in relevant commits swiftly.

Like:

- https://github.com/Automattic/sensei/tree/trunk/assets
- https://github.com/Automattic/sensei/tree/trunk/lang
- https://github.com/Automattic/sensei/tree/trunk/sample-data
- https://github.com/Automattic/sensei/tree/trunk/scripts
- https://github.com/Automattic/sensei/tree/trunk/tests
- https://github.com/Automattic/sensei/tree/trunk/themes
- https://github.com/Automattic/sensei/tree/trunk/includes/block-patterns
- https://github.com/Automattic/sensei/tree/trunk/includes/blocks
- https://github.com/Automattic/sensei/tree/trunk/includes/course-theme

**Comment by saas786** [comment]
@pawelkmpt 

Split PR into two separate PRs.

[this](https://github.com/conversionxl/sensei/pull/9) focuses only sentry log error only, so we can ship it quickly.

New PR is: https://github.com/conversionxl/sensei/pull/10

**Review by pawelkmpt — APPROVED** [review]

**Pull request merged**